aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -2370,6 +2406,25 @@ def _call_exception_handler(self, exp: Exception, route: Route) -> ResponseBuild
                 route=route,
             )
 
+        # OpenAPIValidationMiddleware will only raise ResponseValidationError when
+        # 'self._response_validation_error_http_status' is not None
+        if isinstance(exp, ResponseValidationError):
+            http_status = (
+                self._response_validation_error_http_status
+                if self._response_validation_error_http_status
+                else HTTPStatus.UNPROCESSABLE_ENTITY
+            )
```

Comment:
```suggestion
            http_status = self._response_validation_error_http_status
```
Am I wrong or aren't we testing this condition before setting the HTTP code? Do we need it again here?

Reply:
Absolutely right